src/sdk/trace/sampler.rs

```diff
@@ -28,17 +28,29 @@ impl api::Sampler for Sampler {
     ) -> api::SamplingResult {
         let decision = match self {
             // Always sample the trace
-            Sampler::Always => api::SamplingDecision::RecordAndSampled,
+            Sampler::AlwaysOn => api::SamplingDecision::RecordAndSampled,
             // Never sample the trace
-            Sampler::Never => api::SamplingDecision::NotRecord,
-            // Sample if the parent span is sampled
-            Sampler::Parent => {
-                if parent_context.map(|ctx| ctx.is_sampled()).unwrap_or(false) {
-                    api::SamplingDecision::RecordAndSampled
-                } else {
-                    api::SamplingDecision::NotRecord
-                }
-            }
+            Sampler::AlwaysOff => api::SamplingDecision::NotRecord,
+            // The parent decision if sampled; otherwise the decision of delegate_sampler
+            Sampler::ParentOrElse(delegate_sampler) => parent_context.map_or(
+                delegate_sampler
+                    .should_sample(
+                        parent_context,
+                        trace_id,
+                        _name,
```

Comment:
can we remove these underscores now that the variables are being used?

Reply:
😬 Fixed